plugins/transcend/transcend-nvme.c

```diff
@@ -76,11 +127,288 @@ static int getBadblock(int argc, char **argv, struct command *cmd, struct plugin
 	nvmecmd.addr = (__u64)(uintptr_t)data;
 	nvmecmd.data_len = 0x1;
 	result = nvme_submit_admin_passthru(dev_fd(dev), &nvmecmd, NULL);
-	if (!result) {
-		int badblock  = data[0];
+	if (!result)
+	{
+		int badblock = data[0];
 
 		printf("Transcend NVME badblock count: %d\n", badblock);
 	}
 	dev_close(dev);
 	return result;
 }
+
+const char *format_char_array(char *str, int strsize, unsigned char *chr, int chrsize)
+{
+	int b = 0;
+	while (b < chrsize && chr[b] == ' ')
+		b++;
+	int n = 0;
+	while (b + n < chrsize && chr[b + n])
+		n++;
+	while (n > 0 && chr[b + n - 1] == ' ')
+		n--;
+
+	if (n >= strsize)
+		n = strsize - 1;
+
+	for (int i = 0; i < n; i++)
+	{
```

Comment:
Those fuctions implemented should be follwed the kernel coding style below.
  <https://docs.kernel.org/process/coding-style.html>